src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +14,29 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            session.beginTransaction();
+            session.persist(movie);
+            session.getTransaction().commit();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't add movie: " + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
Consider using try-with-resources for the `Session` in the `add` method to ensure it is closed properly, even if an exception occurs. This will make the code cleaner and more robust.